src/s2/s2region_term_indexer.cc

```diff
@@ -245,18 +278,19 @@ vector<string> S2RegionTermIndexer::GetQueryTermsForCanonicalCovering(
     S2_DCHECK_GE(level, options_.min_level());
     S2_DCHECK_LE(level, options_.max_level());
     S2_DCHECK_EQ(0, (level - options_.min_level()) % options_.level_mod());
+    // assume level <= options_.true_max_level()
 
     // Cells in the covering are always queried as ancestor terms.
-    terms.push_back(GetTerm(TermType::ANCESTOR, id, prefix));
+    terms->push_back(GetTerm(TermType::ANCESTOR, id, prefix));
 
     // If the index only contains points, there are no covering terms.
     if (options_.index_contains_points_only()) continue;
 
     // If we are optimizing for index space rather than query time, cells are
     // also queried as covering terms (except for true_max_level() cells,
     // which are indexed and queried as ancestor cells only).
-    if (options_.optimize_for_space() && level < true_max_level) {
-      terms.push_back(GetTerm(TermType::COVERING, id, prefix));
+    if (options_.optimize_for_space() && level != true_max_level) {
```

Comment:
Only change what you need to. More changes just makes it harder to see what's really changing.

Reply:
I just want to make it same for different function